sdk/search/search-documents/test/public/utils/recordedClient.ts

```diff
@@ -41,34 +34,43 @@ export const testEnv = new Proxy(envSetupForPlayback, {
   },
 });
 
-const serviceName = process.env.ENDPOINT?.match(/https?:\/\/(.*).search.windows.net/)?.[1];
+const generalSanitizers = [];
 
-const sanitizerOptions = serviceName
-  ? {
-      bodySanitizers: [
-        {
-          target: serviceName, // extract subdomain
-          value: "service-name",
-        },
-      ],
-    }
-  : undefined;
+if (env.ENDPOINT) {
+  generalSanitizers.push({
+    regex: false,
+    value: "subdomain",
+    target: env.ENDPOINT.match(/:\/\/(.*).search.windows.net/)![1],
+  });
+}
+
+if (env.OPENAI_ENDPOINT) {
+  generalSanitizers.push({
+    regex: false,
+    value: "subdomain",
+    target: env.OPENAI_ENDPOINT.match(/:\/\/(.*).openai.azure.com/)![1],
+  });
+}
 
 const recorderOptions: RecorderStartOptions = {
   envSetupForPlayback,
-  sanitizerOptions,
+  sanitizerOptions: {
+    generalSanitizers,
+  },
 };
 
-export async function createClients<IndexModel>(
+export async function createClients<IndexModel extends object>(
   serviceVersion: string,
   recorder: Recorder,
   indexName: string
 ): Promise<Clients<IndexModel>> {
   await recorder.start(recorderOptions);
 
   indexName = recorder.variable("TEST_INDEX_NAME", indexName);
-  const endPoint: string = process.env.ENDPOINT ?? "https://endpoint";
+  const endPoint: string = env.ENDPOINT ?? "https://endpoint";
```

Comment:
Could we load all env vars using `assertEnvironmentVariable`?

Reply:
Done!